src/pages/Typing/Typing.tsx

```diff
@@ -0,0 +1,135 @@
+import { Fragment, FC, useState } from 'react'
+import { useSelector, useDispatch } from 'react-redux'
+import { Container, Paper, Grid, TextField } from '@material-ui/core'
+import { TypingHeader, Typography, Result, Loader } from '../../components'
+import { typingStyles } from './Typing.styles'
+import { saveTestResult } from '../../redux'
+
+interface TypingProps {}
+
+const Typing: FC<TypingProps> = () => {
+  const {
+    testData: { testInfo: { paragraph = '', level = '', time = 0 } = {} } = {},
+    userLogin: { userInfo: { displayName = '', photoURL = '' } = {} } = {},
```

Comment:
This level of destructuring is very hard to read and understand. Can it be done in a more readable way?